listeners.go

```diff
@@ -495,6 +495,21 @@ func ParseNetworkAddress(addr string) (NetworkAddress, error) {
 	}, nil
 }
 
+// SplitNetwork splits a into a network, if any, and rest.
+func SplitNetwork(a string) (network, rest string, err error) {
+	beforeSlash, afterSlash, slashFound := strings.Cut(a, "/")
+	if slashFound {
+		network = strings.ToLower(strings.TrimSpace(beforeSlash))
+		a = afterSlash
+	}
+	if isUnixNetwork(network) {
+		rest = a
+		return
+	}
+	rest = a
+	return
```

Comment:
:thinking: 

Couldn't this whole function basically become:

```go
return strings.Cut(a, "/)
```

Thus we can eliminate the function altogether.

Reply:
Almost. The third return value of strings.Cut is a bool, not an error.

But regardless of this detail, I’d argue that even if all the function does is call `strings.Cut`, having it have a descriptive name (`SplitNetwork`) is preferable to eliminating it, for clarity.

Reply:
We never assign the error, do we? I think using `strings.Cut` gives us more information then, since at least the bool tells us something. Plus there's no guesswork of "how is this different from doing strings.Cut" -- because it's not :smiley: (which is a good thing!)